packages/next-sitemap/src/url/util/index.ts

```diff
@@ -18,5 +18,5 @@ export const generateUrl = (baseUrl: string, slug: string): string => {
  * @param path path check
  */
 export const isNextInternalUrl = (path: string): boolean => {
-  return new RegExp(/[^\/]*^.[_]|(?:\[)/g).test(path)
+  return new RegExp(/[^\/]*^.[_]|404$|(?:\[)/g).test(path)
```

Comment:
@trevorblades Thanks for the PR. I do agree to exclude 404 from indexing as it creates crawling issues. 

However your updated regexp treats any url that contains letter 404 as internal urls. Which is not ideal. Check the screenshot for examples

<img width="927" alt="Screenshot 2020-12-30 at 5 07 52 AM" src="https://user-images.githubusercontent.com/4602725/103321030-cdcb5180-4a5d-11eb-9186-6370ca62e0c4.png">

Reply:
Good catch! I just updated the regex to look for `/404` specifically. I could also implement this change differently, depending on what syntax you prefer using:

```js
return path === '/404' || new RegExp(...).test(path)
```

Reply:
Let's stick with regexp for now. Conditional syntax might introduce complexity in future i18n routes support.